tests/ras/test_main_c.cpp

```diff
@@ -0,0 +1,704 @@
+  struct RASCommandLine cmd_line_;
+  test_platform platform_;
+  test_system *system_;
+};
+
+/**
+ * @test       main_params_01
+ * @brief      Test: ras_main
+ * @details    When ras_main is called with an valid command,<br>
+ *             that includes errors injection, ras_main returns zero.<br>
+ */
+TEST_P(ras_c_mock_p, main_params_01){
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  char four[20];
+  char five[20];
+  char six[20];
+  char seven[20];
+  char eight[20];
+  char nine[20];
+  char ten[20];
+  char eleven[20];
+  char twelve[20];
+  char thirteen[20];
+  char fourteen[20];
+
+  strcpy(zero, "ras");
+  strcpy(one, "--segment");
+  sprintf(two, "%d", platform_.devices[0].segment);
+  strcpy(three, "-B");
+  sprintf(four, "%d", platform_.devices[0].bus);
+  strcpy(five, "-D");
+  sprintf(six, "%d", platform_.devices[0].device);
+  strcpy(seven, "-F");
+  sprintf(eight, "%d", platform_.devices[0].function);
+  strcpy(nine, "-P");
+  strcpy(ten, "-Q");
+  strcpy(eleven, "-R");
+  strcpy(twelve, "-O");
+  strcpy(thirteen, "-N");
+  strcpy(fourteen, "-C");
+
+  char *argv[] = { zero, one, two, three, four,
+                   five, six, seven, eight, nine,
+                   ten, eleven, twelve, thirteen,
+                   fourteen, NULL };
+
+  EXPECT_EQ(ras_main(15, argv), FPGA_OK);
+}
+
+/**
+ * @test       main_params_03
+ * @brief      Test: ras_main
+ * @details    When ras_main is called with an valid command option,<br>
+ *             injecting mmio error, ras_main returns zero.<br>
+ */
+TEST_P(ras_c_mock_p, main_params_03){
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  char four[20];
+  char five[20];
+  char six[20];
+  char seven[20];
+  char eight[20];
+
+  strcpy(zero, "ras");
+  strcpy(one, "-B");
+  sprintf(two, "%d", platform_.devices[0].bus);
+  strcpy(three, "-P");
+  strcpy(four, "-E");
+  strcpy(five, "-G");
+  strcpy(six, "-H");
+  strcpy(seven, "-I");
+  strcpy(eight, "-C");
+
+  char *argv[] = { zero, one, two, three, four,
+                   five, six, seven, eight, NULL };
+
+  EXPECT_EQ(ras_main(9, argv), FPGA_OK);
+}
+
+/**
+ * @test       mmio_error
+ * @brief      Test: test_mmio_error 
+ * @details    When mmio_error is called with an valid afu handle<br>
+ *             and RAS command to inject, it returns FPGA_OK.<br>
+ */
+TEST_P(ras_c_mock_p, test_mmio_error){
+  cmd_line_ = { 0, false,
+                false, false, false,false,
+                false, false, true, true, false};
+
+  EXPECT_EQ(FPGA_INVALID_PARAM, mmio_error(handle_accel_, nullptr));
+
+  EXPECT_EQ(FPGA_OK, mmio_error(handle_accel_, &cmd_line_));
+}
+
+/**
+ * @test       mmio_errors
+ * @brief      Test: invalid_mmio_errors
+ * @details    When given an invalid handle, mmio_errors returns <br>
+ *             FPGA_INVALID_PARAM.<br>
+ */
+TEST_P(ras_c_mock_p, invalid_mmio_errors){
+  EXPECT_EQ(FPGA_INVALID_PARAM, mmio_error(nullptr, &cmd_line_)); 
```

Comment:
I think this test will fail, since this is the new code which I had difficulty with.  Do we not run the legacy tests in CI?